### ISSUE TITLE
oAuth authentication for Bitbucket private repos stopped working with v1.3.0 

### ISSUE THREAD

**Issue by acelaya** [composer/composer#5996]
My `composer.json`:

```json
{
    "name": "future/rest-api",
    "license": "proprietary",
    "require": {
        "php": "^7.0",
        "zendframework/zend-expressive": "^1.0",
        "zendframework/zend-stdlib": "^3.0",
        "zendframework/zend-expressive-fastroute": "^1.3",
        "zendframework/zend-servicemanager": "^3.0",
        "zendframework/zend-expressive-twigrenderer": "^1.0",
        "zendframework/zend-config": "^2.6",
        "zendframework/zend-i18n": "^2.7",
        "zendframework/zend-inputfilter": "^2.7",
        "zendframework/zend-permissions-acl": "^2.6",
        "zendframework/zend-crypt": "^3.1",
        "zendframework/zend-paginator": "^2.7",
        "zendframework/zend-config-aggregator": "^0.1.0",
        "acelaya/zsm-annotated-services": "^1.0",
        "doctrine/orm": "^2.5",
        "doctrine/migrations": "^1.4",
        "symfony/console": "^3.1",
        "symfony/process": "^3.1",
        "firebase/php-jwt": "^4.0",
        "monolog/monolog": "^1.21",
        "theorchard/monolog-cascade": "^0.4",
        "future/common": "^2.2",
        "future/core": "^2.1",
        "swiftmailer/swiftmailer": "^5.4",
        "myclabs/php-enum": "^1.4",
        "ramsey/uuid": "^3.5",
        "ramsey/uuid-doctrine": "^1.2",
        "league/fractal": "^0.14",
        "predis/predis": "^1.1",
        "guzzlehttp/guzzle": "^6.2",
        "jms/serializer": "^1.4"
    },
    "require-dev": {
        "phpunit/phpunit": "^5.0",
        "phpunit/dbunit": "^2.0",
        "phpunit/phpcov": "^3.1",
        "squizlabs/php_codesniffer": "^2.3",
        "filp/whoops": "^2.0",
        "symfony/var-dumper": "^3.1",
        "vlucas/phpdotenv": "^2.3",
        "devster/ubench": "^1.2"
    },
    "autoload": {
        "psr-4": {
            
        },
        "files": [
            
        ]
    },
    "autoload-dev": {
        "psr-4": {
            
        }
    },
    "scripts": {
        "check": [
            "@cs",
            "@test"
        ],
        "ci": [
            "@check",
            "@func-test"
        ],
        "cs": "phpcs",
        "cs-fix": "phpcbf",
        "test": "phpunit --colors=always",
        "func-test": "phpunit --colors=always --config=phpunit-func.xml",
        "pretty-test": "phpunit --coverage-html build/html",
        "complete-pretty-test": [
            "phpunit --coverage-php build/coverage-unit.cov",
            "phpunit --config=phpunit-func.xml --coverage-php build/coverage-func.cov",
            "phpcov merge --html build/html build"
        ],
        "migrate": "doctrine-migrations migrations:migrate --no-interaction",
        "migrate-prev": "doctrine-migrations migrations:migrate prev --no-interaction"
    },
    "config": {
        "process-timeout": 0
    },
    "repositories": [
        {
            "type": "vcs",
            "url": "https://bitbucket.org/future/common.git"
        },
        {
            "type": "vcs",
            "url": "https://bitbucket.org/future/core.git"
        }
    ]
}
```

Output of `composer diagnose`:

```
Checking composer.json: OK
Checking platform settings: OK
Checking git settings: OK
Checking http connectivity to packagist: OK
Checking https connectivity to packagist: OK
Checking github.com rate limit: OK
Checking disk free space: OK
Checking pubkeys: 
Tags Public Key Fingerprint: 57815BA2 7E54DC31 7ECC7CC5 573090D0  87719BA6 8F3BB723 4E5D42D0 84A14642
Dev Public Key Fingerprint: 4AC45767 E5EC2265 2F0C1167 CBBB8A2B  0C708369 153E328C AD90147D AFE50952
OK
Checking composer version: OK
```

When I run this command:

```
composer update
```

I get the following output:

```
Loading composer repositories with package information

                                                
  [Composer\Downloader\TransportException]      
  Could not authenticate against bitbucket.org  
                                                

update [--prefer-source] [--prefer-dist] [--dry-run] [--dev] [--no-dev] [--lock] [--no-custom-installers] [--no-autoloader] [--no-scripts] [--no-progress] [--no-suggest] [--with-dependencies] [-v|vv|vvv|--verbose] [-o|--optimize-autoloader] [-a|--classmap-authoritative] [--apcu-autoloader] [--ignore-platform-reqs] [--prefer-stable] [--prefer-lowest] [-i|--interactive] [--root-reqs] [--] [<packages>]...
```

And I expected this to happen:

To update the components.

We have two private bitbucket repositories, to which we authenticate using oAuth keys defined in the `~/.composer/auth.json` file.
It was properly working until we have updated to composer v1.3.0. With this version we get the previous error.
However, it works while using plain username and password instead of oAuth keys, but that's not an option for our continuous integration server

**Comment by margori** [comment]
I'm also having this issue.

**Comment by Seldaek** [comment]
Fixed by #5993, use `composer self-update --snapshot` for now to get the latest version.
